crates/types/src/consensus.rs

```diff
@@ -406,6 +406,8 @@ pub struct CommitmentAndMetadata<TYPES: NodeType> {
     pub metadata: <TYPES::BlockPayload as BlockPayload>::Metadata,
     /// Builder fee data
     pub fee: BuilderFee<TYPES>,
+    /// View number this block is for
+    pub block_view: TYPES::Time,
```

Comment:
Quick question about this: how does this differ from `cur_view` and does this clobber that value? Like, will these values ever differ?

Reply:
My understanding is that we request 1 view in advance, so `cur_view` isn't really descriptive. I think `block_view` is more accurate; we're requesting a block to propose with in a specific (future) view. But I'm not that familiar with this code so there might be some edge cases here that I'm missing.

(Also note that the struct does not have a `cur_view`.)